### PR TITLE
Implement Slices

### PR DESCRIPTION
**Pull request by tobiasstrebitzer**
Extract slice information as described in the [spec](https://github.com/aseprite/aseprite/blob/master/docs/ase-file-specs.md#slice-chunk-0x2022).
[Check out PR to implement slices](https://app.gitkraken.com/glo/card/527439ea16f24d909f588d5b1c032e3f)

### CODE REVIEW

**Comment by TheCyberRonin** [comment]
I'm so sorry, the notification for this must have gotten buried in my others 😭 . I'll check this out and give it some testing!

I appreciate the work, sorry again 😓

File: Aseprite.js [hunk 1]
```diff
@@ -224,6 +227,34 @@ class Aseprite {
     this.colorDepth === 8 ? palette.index = this.paletteIndex : '';
     return palette;
   }
+  readSliceChunk() {
+    const numSliceKeys = this.readNextDWord();
+    const flags = this.readNextDWord();
+    const patch = (flags & 1) !== 0 ? {} : null
+    const pivot = (flags & 2) !== 0 ? {} : null
```
Comment (TheCyberRonin) [inline Aseprite.js:234]: If the patch and pivot data aren't present, it just shouldn't be a part of the `slice` property.

File: Aseprite.js [hunk 2]
```diff
@@ -224,6 +227,34 @@ class Aseprite {
...
+    this.skipBytes(4);
+    const name = this.readNextString()
+    const keys = []
+    for(let i = 0; i < numSliceKeys; i ++) {
+      const frameNumber = this.readNextDWord();
```
Comment (TheCyberRonin) [inline Aseprite.js:239]: Could remedy the patch and pivot issue by maybe adding an object in this loop to be pushed onto the slices property so you can decide if it needs added or not.

File: typings/Aseprite.d.ts [hunk 3]
```diff
@@ -66,6 +67,23 @@ declare namespace Aseprite {
...
+    patch?: {
+      x: number;
+      y: number;
+      width: number;
+      height: number;
+    };
+    pivot?: {
+      x: number;
+      y: number;
+    }
```
Comment (doctorpangloss) [inline typings/Aseprite.d.ts:85]: This is missing `name`

**Comment by doctorpangloss** [comment]
The types are wrong, objects look like:

```
{"flags":0,"name":"....","keys":[{"frameNumber":0,"x":8,"y":19,"width":35,"height":54,"patch":null,"pivot":null}]}
```

File: Aseprite.js (hunk 2 above)
Comment (tobiasstrebitzer) [reply Aseprite.js:237]: Got it, fixed with a59c8a6

**Comment by tobiasstrebitzer** [comment]
Updated assignment of `pivot`, `patch` and fixed typings.

**Comment by doctorpangloss** [comment]
It turns out there are other issues with this library where it will just cause node to crash :(

Sorry my dude

**Comment by tobiasstrebitzer** [comment]
@doctorpangloss it's working pretty reliable for me, been using it for a while now, with rather complex aseprite documents. I'm using node v12.19.0.

Happy to help if you got a stacktrace/ error logs.

**Comment by doctorpangloss** [comment]
It looks like the underlying problem is that the specs from https://github.com/aseprite/aseprite/blob/master/docs/ase-file-specs.md do not match the actual saved file here: https://github.com/aseprite/aseprite/blob/master/src/doc/slice_io.cpp

**Review by TheCyberRonin — APPROVED** [review]
Seems good, thanks for maing those changes. Will merge in today and try to use the change for a GitHub action to automatically push to npm! Appreciate the work!!

**Comment by TheCyberRonin** [comment]
> It turns out there are other issues with this library where it will just cause node to crash :(
> 
> Sorry my dude

Maybe you're using a very complex Aseprite file or something? I'm not sure, but I haven't had Node crash on me using the library once yet. If you're having problems with it, please pull up an issue and I'll see if I can address it!

**Comment by TheCyberRonin** [comment]
Closes #6

**Pull request merged**